### DISCUSSION TITLE
web端使用代码播放pag文件掉帧

### DISCUSSION THREAD

**Discussion by hht123-hht** [Tencent/libpag#2402] [Q&A] [answered]
真实动画视频
https://github.com/user-attachments/assets/e11f214e-3eb9-414a-8d45-d28964e484f6

但是通过web端代码播放该pag文件会掉帧，主要表现在手拿起物品后未在空中悬停1s
https://github.com/user-attachments/assets/d7564c7b-785f-49a5-8049-72fa1c8872c3

pag文件
[tiphand_b.zip](https://github.com/user-attachments/files/16504422/tiphand_b.zip)

**Comment by mianma12138** [comment]
我也遇到了。pag播放大型动效的时候，会出现卡顿，严重掉帧的现象。

**Comment by kevingpqi123** [comment, marked as answer]
大家多看下 pag 官网上面的文章：https://pag.art/docs/optimization.html
这里的动效其实不复杂，采用纯矢量的实现方式可以获得很好的性能，BMP 预合成本质上就是视频，在 Web 端视频解码始终是性能瓶颈。
后续大家遇到性能问题，首先需要分析的就是素材，不是所有的 pag 的文件的性能都是一样，而是和素材复杂度强相关。
